release/models/system/openconfig-connections.yang

```diff
@@ -0,0 +1,247 @@
+module openconfig-connections {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/system/connections";
+
+  prefix "oc-conn";
+
+  // import some basic types
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-inet-types { prefix oc-inet; }
+  import openconfig-system { prefix oc-sys; }
+  import openconfig-network-instance { prefix oc-ni; }
+
+  // meta
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module provides data definitions for network connections.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2025-08-13" {
+    description
+      "Initial public release";
+    reference "0.1.0";
+  }
+
+  // OpenConfig specific extensions for module metadata.
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // grouping statements
+
+  grouping connections-top {
+    description
+      "Top level grouping for attributes for network connections.";
+
+    container connections {
+      description
+        "Parameters related to all active network connections";
+
+      list connection {
+        key "network-instance socket";
+        config false;
+        description
+          "Parameters related to individual active network connections";
+
+        leaf network-instance {
```

Comment:
What does this leaf point to? In the context of Ubuntu/Linux (since that is 1 of 2 implementation listed in the PR description) does it point to the network interface that a socket is _explicitly_ bound to, or the interface a route for the socket is cached to egress through? 

What if the socket is bound to a specific local address but no remote, what should show up here? Similarly, what if the socket is bound all addresses?